From 9fd3c733c645be13a2e39a3bae3983f88530b48d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lukas=20M=C3=A4rdian?= <lukas.maerdian@canonical.com>
Date: Thu, 7 Jan 2021 12:46:22 +0100
Subject: [PATCH] zipl/boot/Makefile: -no-pie is not a valid ld flag

The "-no-pie" option never has been a valid flag of ld. It breaks the build with newer binutils.
See:
https://sourceware.org/bugzilla/show_bug.cgi?id=27050
https://bugs.launchpad.net/ubuntu/+source/s390-tools/+bug/1907789

[Retrieved from: https://github.com/ibm-s390-linux/s390-tools/pull/106]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 zipl/boot/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zipl/boot/Makefile b/zipl/boot/Makefile
index a2e5f190..b81df778 100644
--- a/zipl/boot/Makefile
+++ b/zipl/boot/Makefile
@@ -109,7 +109,7 @@ stage3.bin:	stage3.exec
 		$< $@
 
 data.o: $(FILES)
-	$(LD) $(NO_PIE_LDFLAGS) -r -b binary -o data.o $(FILES)
+	$(LD) -r -b binary -o data.o $(FILES)
 
 data.h: data.o
 	rm -f data.h
